server/svix-server/src/v1/endpoints/attempt.rs

```diff
@@ -65,6 +65,101 @@ impl From<messageattempt::Model> for MessageAttemptOut {
     }
 }
 
+// TODO: Mostly similar to `v1::endpoints::message::MessageOut, is there some way to reduce
+// duplication cleanly?
+
+/// A model containing information on a given message plus additional fields on the last attempt for
+/// that message.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, ModelOut, FromQueryResult)]
+#[serde(rename_all = "camelCase")]
+struct AttemptedMessageOut {
+    event_type: EventTypeName,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_id: Option<MessageUid>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    channels: Option<EventChannelSet>,
+    payload: serde_json::Value,
+    id: MessageId,
+    timestamp: DateTimeWithTimeZone,
+    status: MessageStatus,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    next_attempt: Option<DateTimeWithTimeZone>,
+}
+
+impl AttemptedMessageOut {
+    pub fn from_message_and_destination(
+        message: message::Model,
+        destination: messagedestination::Model,
+    ) -> Self {
+        Self {
+            event_type: message.event_type,
+            event_id: message.uid,
+            channels: message.channels,
+            payload: message.payload,
+            id: message.id,
+            timestamp: message.created_at,
+            status: destination.status,
+            next_attempt: destination.next_attempt,
+        }
+    }
+}
+
+/// Additional parameters (besides pagination) in the query string for the "List Attempted Messages"
+/// enpoint.
+#[derive(Debug, Deserialize, Validate)]
+pub struct ListAttemptedMessagesQueryParameters {
+    #[validate]
+    channel: Option<EventChannel>,
+    status: Option<MessageStatus>,
+}
+
+/// GET /api/v1/app/:app_id/enpoint/:endp_id/msg/
```

Comment:
I'm not a fan of this, we don't add this comment elsewhere, and it's nice that the routing is handled in one place, this makes it disjoin.